plugins/modules/accesspoint_workflow_manager.py

```diff
@@ -0,0 +1,818 @@
+        (device_exists, current_ap_config) = self.get_current_config(input_config)
+        self.log("Current AP config details (have): {0}".format(str(current_ap_config)), "DEBUG")
+        if device_exists:
+            self.have["mac_address"] = current_ap_config.get("mac_address")
+            self.have["device_exists"] = device_exists
+            self.have["current_ap_config"] = current_ap_config
+        self.log("Current State (have): {0}".format(str(self.have)), "INFO")
+        return self
+
+    def get_diff_merged(self, config):
+        """
+        Update/Create wireless accesspoint configuration in Cisco Catalyst Center with fields
+        provided in the playbook.
+        Parameters:
+          self (object): An instance of a class used for interacting with Cisco Catalyst Center.
+          config (dict): A dictionary containing configuration information.
+        Returns:
+            self (object): An instance of a class used for interacting with Cisco Catalyst Center.
+        Description:
+            This method determines whether to update or create AP configuration in Cisco
+            Catalyst Center based on the provided configuration information. 
+            If the specified site exists, the method checks if it requires an update
+            by calling the 'update_ap_configuration' method. If an update is required, 
+            it calls the 'configure_access_points' function from the 'wireless' family of 
+            the Cisco Catalyst Center API. If Current configuration same as input configuration 
+            does not require an update, the method exits, indicating that Accesspoint 
+            configuration is up to date.
+        """
+
+        config_updated = False
+        config_created = False
+        task_response = None
+        # check if the given AP config exists and/or needs to be updated/created.
+
+        if self.have.get("device_exists"):
+            consolidated_data = self.compare_ap_cofig_with_inputdata(self.have["current_ap_config"])
+            if consolidated_data:
+                self.log('Final AP Configuration data to update {}'.format(str(consolidated_data)),
+                      "INFO")
+                task_response = self.update_ap_configuration(consolidated_data)
+                self.log('Task respoonse {}'.format(str(task_response)),"INFO")
+                config_updated = True
+            else:
+                # Accesspoint does not need update
+                self.msg = "AP - {0} does not need any update"\
+                    .format(self.have.get("current_ap_config").get("ap_name"))
+                self.log(self.msg, "INFO")
+                responses = {}
+                del self.payload["device_list"]
+                responses["accesspoints_updates"] = {"response": self.payload["device_config"]}
+                self.result['msg'] = self.msg
+                self.result["response"].append(responses)
+                self.result["skipped"] = True
```

Comment:
self.result['changed'] = False 
As it doesnot need any update.

Reply:
changed